### PR TITLE
Bug fixes for Salesforce Network SDK

### PR DESCRIPTION
**Pull request by wmathurin**

### CODE REVIEW

File: SalesforceNetworkSDK/SFNetworkEngine+Internal.h [hunk 1]
```diff
@@ -67,9 +67,9 @@
  */
 - (void)replayOperationsWaitingForNetwork;
 
-/**Clone an operation. Used to re-queue a failed operation
+/**Clone the internal operation. Used to re-queue a failed operation
```
Comment (wmathurin) [inline SalesforceNetworkSDK/SFNetworkEngine+Internal.h:70]: Cloning the SFNetworkOperation was bogus, the new SFNetworkOperation would contain a new MKNetworkOperation with completion/error blocks copied from the old SFNetworkOperation and therefore executing with the old SFNetworkOperation as context.

File: SalesforceNetworkSDK/SFNetworkOperation.m [hunk 2]
```diff
@@ -280,7 +280,7 @@ -(BOOL) canCallback {
 - (void)addCompletionBlock:(SFNetworkOperationCompletionBlock)completionBlock errorBlock:(SFNetworkOperationErrorBlock)errorBlock{
     if (_internalOperation) {
         __weak SFNetworkOperation *weakSelf = self;
-        [_internalOperation onCompletion:^(MKNetworkOperation *completedOperation) {
+        [_internalOperation addCompletionHandler:^(MKNetworkOperation *completedOperation) {
```
Comment (wmathurin) [inline SalesforceNetworkSDK/SFNetworkOperation.m:283]: onCompletion is deprecated (and is actually broken)

File: SalesforceNetworkSDK/SFNetworkOperation.m [hunk 3]
```diff
@@ -313,15 +313,6 @@ - (void)addCompletionBlock:(SFNetworkOperationCompletionBlock)completionBlock er
...
             if (errorBlock) {
-                if ([SFNetworkUtils typeOfError:error] == SFNetworkOperationErrorTypeAccessDenied) {
-                    //Permission denied error
-                    NSError *potentialError = [weakSelf checkForErrorInResponse:weakSelf.internalOperation];
-                    
-                    if (potentialError) {
-                        error = potentialError;
-                    }
-                }
-
```
Comment (wmathurin) [inline SalesforceNetworkSDK/SFNetworkOperation.m:324]: This is pointless, we have already checked for the error in the response a few lines earlier (line 299).

File: SalesforceNetworkSDK/SFNetworkOperation.m [hunk 4]
```diff
@@ -485,22 +476,11 @@ - (void)callDelegateDidFailWithError:(NSError *)error {
...
     if (weakSelf.delegate) {
-        if ([SFNetworkUtils typeOfError:error] == SFNetworkOperationErrorTypeAccessDenied) {
-            //Permission denied error
-            //Server side sometimes return 403 with JSON object to
-            //indicate permission denied error
-            NSError *potentialError = [weakSelf checkForErrorInResponseStr:self.responseAsString withError:error];
-            if (potentialError) {
-                error = potentialError;
-            }
-        }
```
Comment (wmathurin) [inline SalesforceNetworkSDK/SFNetworkOperation.m:496]: This is pointless, we have already checked for the error in the response a few lines earlier.

File: SalesforceNetworkSDK/SFNetworkOperation.m [hunk 5]
```diff
@@ -485,22 +476,11 @@ - (void)callDelegateDidFailWithError:(NSError *)error {
...
         if (error.code == kCFURLErrorTimedOut) {
             if ([weakSelf.delegate respondsToSelector:@selector(networkOperationDidTimeout:)]) {
                 if([self canCallback]) {
                     dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
-                        if (error.code == kCFURLErrorTimedOut) {
-                            [weakSelf.delegate networkOperationDidTimeout:weakSelf];
-                        }
+                        [weakSelf.delegate networkOperationDidTimeout:weakSelf];
```
Comment (wmathurin) [inline SalesforceNetworkSDK/SFNetworkOperation.m:483]: We are already inside a if ( time out ) [see line 479]

File: SalesforceNetworkSDK/SFNetworkOperation.m [hunk 6]
```diff
@@ -511,9 +491,7 @@ - (void)callDelegateDidFailWithError:(NSError *)error {
         if ([weakSelf.delegate respondsToSelector:@selector(networkOperation:didFailWithError:)]) {
             if([self canCallback]) {
                 dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_BACKGROUND, 0), ^{
-                    if (error.code == kCFURLErrorTimedOut) {
-                        [weakSelf.delegate networkOperation:weakSelf didFailWithError:error];
-                    }
+                    [weakSelf.delegate networkOperation:weakSelf didFailWithError:error];
```
Comment (wmathurin) [inline SalesforceNetworkSDK/SFNetworkOperation.m:494]: The time out case is handled by the previous if

**Comment by khawkins** [comment]
These updates look fine to me.

**Pull request merged**